### PR TITLE
VL-120_Generate-id-for-table-rows_Dmytro-Holdobin

### PR DESCRIPTION
**Pull request by Explicit12**
Table adds id to row if necessary

> https://ilevel.atlassian.net/browse/VL-116

### CODE REVIEW

File: src/ui.data-table/UTable.vue [hunk 1]
```diff
@@ -633,9 +627,10 @@ function setHeaderCellWidth() {
   const mainHeaderItems = [...(headerRowRef.value?.children || [])];
   const stickyHeaderItems = [...(stickyHeaderRowRef.value?.children || [])];
 
-  stickyHeaderItems.forEach((item, index) => {
-    item.style.width = `${mainHeaderItems[index]?.offsetWidth / PX_IN_REM}rem`;
-  });
+  stickyHeaderItems.map((item, index) => ({
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/UTable.vue:630]: map retuns a new array. So where is you use it?

File: src/ui.data-table/utilTable.js [hunk 2]
```diff
@@ -15,13 +17,29 @@ export function getFilteredRow(row, columns) {
...
-  if (row.row) {
+  if (row.row && !Array.isArray(row.row)) {
     row.row = syncRowCheck(row.row, selectedRows);
   }
 
   return row;
 }
 
+export function addRowId(row) {
+  const isRowId = typeof row.id !== "undefined" && row.id !== null && row.id !== "";
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/utilTable.js:28]: isRowId >>> hasRowId

File: src/ui.data-table/utilTable.js [hunk 3]
```diff
@@ -15,13 +17,29 @@ export function getFilteredRow(row, columns) {
...
+
+  row.id = isRowId ? row.id : getRandomId();
+
+  if (row.row && !Array.isArray(row.row)) {
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/utilTable.js:32]:
```js
if (row.row) {
  row.row = Array.isArray(row.row) 
    ? row.row.map((nestedRow) => addRowId(nestedRow)) 
    : addRowId(row.row);
}
```

File: src/ui.data-table/UTable.vue [hunk 4]
```diff
@@ -621,10 +614,11 @@ function setFooterCellWidth(width) {
   const mainFooterItems = [...footerRowRef.value.children];
   const stickyFooterItems = [...stickyFooterRowRef.value.children];
 
-  stickyFooterItems.forEach((item, index) => {
-    item.style.width =
-      width === null ? `${ZERO_WIDTH}rem` : `${mainFooterItems[index].offsetWidth / PX_IN_REM}rem`;
-  });
+  stickyFooterItems.map((item, index) => ({
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/UTable.vue:617]: map retuns a new array. So where is you use it?

File: src/ui.data-table/utilTable.js (hunk 3 above)
Comment (Explicit12) [reply src/ui.data-table/utilTable.js:32]: That's just condition inside of another condition. Two separate if blocks looks muck cleaner and readable.

File: src/ui.data-table/UTable.vue (hunk 1 above)
Comment (Explicit12) [reply src/ui.data-table/UTable.vue]: Returned as it was before. We manipulate with DOM here, so we don't need any return value.

File: src/ui.data-table/utilTable.js (hunk 3 above)
Comment (itsJohnnyGrid) [reply src/ui.data-table/utilTable.js:32]: `!Array.isArray(row.row)` and `Array.isArray(row.row)` in different conditions looks confusing to me. I prefer to keep conditon and ternary

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**